test/IntegrationTests/MSTest.Acceptance.IntegrationTests/AssemblyAndClassInitializeTimeout.cs

```diff
@@ -0,0 +1,131 @@
+
+using Microsoft.Testing.Platform.Acceptance.IntegrationTests;
+using Microsoft.Testing.Platform.Acceptance.IntegrationTests.Helpers;
+
+namespace MSTest.Acceptance.IntegrationTests;
+
+[TestGroup]
+public class AssemblyAndClassInitializeTimeout : AcceptanceTestBase
+{
+    private readonly string _assetName = "AssemblyAndClassInitializeTimeout";
+    private readonly AcceptanceFixture _acceptanceFixture;
+
+    public AssemblyAndClassInitializeTimeout(ITestExecutionContext testExecutionContext, AcceptanceFixture acceptanceFixture)
+        : base(testExecutionContext)
+    {
+        _acceptanceFixture = acceptanceFixture;
+    }
+
+    [ArgumentsProvider(nameof(GetBuildMatrixTfmBuildConfiguration))]
+    public async Task AssemblyInitialize_Timeout_Should_Stop_Execution(string tfm, BuildConfiguration buildConfiguration)
+    {
+        using TestAsset testAsset = await TestAsset.GenerateAssetAsync(_assetName, SourceCode
+            .PatchCodeWithReplace("$TargetFramework$", tfm)
+            .PatchCodeWithReplace("$MSTestVersion$", MSTestVersion), addPublicFeeds: true);
+        DotnetMuxerResult result = await DotnetCli.RunAsync($"build -nodeReuse:false {testAsset.TargetAssetPath} -c {buildConfiguration}", _acceptanceFixture.NuGetGlobalPackagesFolder.Path);
+        var testHost = TestHost.LocateFrom(testAsset.TargetAssetPath, _assetName, tfm, buildConfiguration: buildConfiguration);
+
+        foreach (string initMethodToTimeout in new[] { "TIMEOUT_CLASSINIT", "TIMEOUT_ASSEMBLYINIT", "TIMEOUT_BASE_CLASSINIT" })
+        {
+            var testhostResult = await testHost.ExecuteAsync(environmentVariables: new Dictionary<string, string>() { { initMethodToTimeout, "1" } });
+
+            if (initMethodToTimeout == "TIMEOUT_CLASSINIT")
+            {
+                testhostResult.AssertOutputContains("Class Initialization method TestClass.ClassInit threw exception. System.Threading.Tasks.TaskCanceledException: A task was canceled..");
+                continue;
+            }
+
+            if (initMethodToTimeout == "TIMEOUT_ASSEMBLYINIT")
+            {
+                testhostResult.AssertOutputContains("Assembly Initialization method TestClass.AssemblyInit threw exception. System.Threading.Tasks.TaskCanceledException: A task was canceled.. Aborting test execution.");
+                continue;
+            }
+
+            if (initMethodToTimeout == "TIMEOUT_BASE_CLASSINIT")
+            {
+                testhostResult.AssertOutputContains("Class Initialization method TestClass.ClassInitBase threw exception. System.Threading.Tasks.TaskCanceledException: A task was canceled..");
+                continue;
+            }
+        }
+    }
+
+    protected const string SourceCode = """
+#file AssemblyAndClassInitializeTimeout.csproj
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <EnableMSTestRunner>true</EnableMSTestRunner>
+    <PlatformTarget>x64</PlatformTarget>
```

Comment:
Is this needed?

Reply:
nope copy paste from past tests where on netfx we got x86 but looks like no more the case.